### PR TITLE
Add Xtensa arch support

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

https://github.com/capstone-engine/capstone/pull/2533
https://github.com/espressif/llvm-project

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Comment by XVilka** [comment]
Probably makes sense to rebase on top of the https://github.com/rizinorg/rizin/pull/4662 to avoid conflicts and double work. cc @Rot127

**Comment by Rot127** [comment]
Yes, though let's wait until the Capstone PRs are merged.

File: librz/arch/p/analysis/analysis_xtensa_cs.c
Comment (XVilka) [inline librz/arch/p/analysis/analysis_xtensa_cs.c]: Either implement or remove?

File: librz/arch/p/analysis/analysis_xtensa_cs.c
Comment (XVilka) [inline librz/arch/p/analysis/analysis_xtensa_cs.c]: Return types do not match

File: test/db/analysis/xtensa
Comment (XVilka) [inline test/db/analysis/xtensa]: Please don't forget to fix it eventually

File: test/db/cmd/cmd_list
Comment (XVilka) [inline test/db/cmd/cmd_list]: Add yourself as the author maybe?

File: librz/arch/isa/xtensa/xtensa.c
Comment (XVilka) [inline librz/arch/isa/xtensa/xtensa.c]: Missing `{}`

File: librz/arch/isa/xtensa/xtensa.h
Comment (XVilka) [inline librz/arch/isa/xtensa/xtensa.h]: Could use just `RZ_XTENSA_H` maybe?

**Review by XVilka — CHANGES_REQUESTED** [review]
I didn't review the ESIL file since I hope we remove it soon. As long as it doesn't break anything, everything is fine in that file.

**Review by XVilka — CHANGES_REQUESTED** [review]
One minor thing left, apart from that - LGTM:
```
[XX] 271 ms   db/cmd/cmd_list Print the asm/analysis plugins
RZ_NOPLUGINS=1 /usr/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -eflirt.sigdb.load.home=false -N -qc La =
-- stdout
--- expected
+++ actual
@@ -62,5 +62,5 @@
 a____ 16 32 64   x86.nz      LGPL3   x86 handmade assembler
 _dA__ 16         xap         PD      XAP4 RISC (CSR)
 _dA__ 32         xcore       BSD     Capstone XCore disassembler (by pancake)
-_dAeI 32         xtensa      LGPL3   Capstone Xtensa disassembly plugin (by billow)
+_dAe_ 32         xtensa      LGPL3   Capstone Xtensa disassembly plugin (by billow)
 adA__ 8          z80         GPL3    Zilog Z80 (by condret)
 ```

**Comment by XVilka** [comment]
OK, LGTM now, just lets wait first for the Capstone PR to be merged first.

File: librz/arch/p/analysis/analysis_xtensa_cs.c
Comment (Rot127) [inline librz/arch/p/analysis/analysis_xtensa_cs.c:39]: Add the return register as well please. `=R0 a2`?

File: librz/arch/p/analysis/analysis_xtensa_cs.c
Comment (Rot127) [inline librz/arch/p/analysis/analysis_xtensa_cs.c]: Are `ReadWrite` values possible? If yes, make it two if cases please.

Comment (Rot127) [reply librz/arch/p/analysis/analysis_xtensa_cs.c]: Ah, below you check for both. So please split it here.

**Review by Rot127 — CHANGES_REQUESTED** [review]
Looks awesome! Just those few things and it lgtm.

**Review by Rot127 — APPROVED** [review]

**Pull request merged**
